wallet/reference/non-evm-apis/starknet-snap-api.md

```diff
@@ -0,0 +1,830 @@
+  "unit": "wei",
+  "includeDeploy": true
+}
+```
+
+</TabItem>
+</Tabs>
+
+## `starkNet_estimateFee`
+
+Gets the estimated gas fee for calling a method on any contract.
+
+### Parameters
+
+- `contractAddress`: `string` - Address of the target contract.
+- `contractFuncName`: `string` - Target function name of the contract.
+- `contractCallData`: `string` - (Optional) Call data for the target function with `,` as a separator.
+- `senderAddress`: `string` - Address of the sender.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Goerli testnet.
+
+### Returns
+
+The response from Starknet's `feeder_gateway/estimate_fee` API endpoint.
+
+### Example
+
+<Tabs>
+<TabItem value="Request">
+
+```js
+await window.ethereum.request({
+  method: "wallet_invokeSnap",
+  params: {
+    snapId: "npm:@consensys/starknet-snap",
+    request: {
+      method: "starkNet_estimateFee",
+      params: {
+        contractAddress: "0x5B38Da6a701c568545dCfcB03FcB875f56beddC4",
+        contractFuncName: "transfer",
+        contractCallData: "0x456...,0x789...,100",
+        senderAddress: "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+        chainId: "0x1"
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+{
+  "suggestedMaxFee": "1000000000000000",
+  "overallFee": "900000000000000",
+  "gasConsumed": "1000000",
+  "gasPrice": "1000000000",
+  "unit": "wei",
+  "includeDeploy": false
+}
+```
+
+</TabItem>
+</Tabs>
+
+## `starkNet_extractPublicKey`
+
+:::note
+This method is integrated into `get-starknet`.
+:::
+
+Extracts the public key from a Starknet account address.
+
+### Parameters
+
+- `userAddress`: `string` - Address of the account contract.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Goerli testnet.
+
+### Returns
+
+The public key of the given account address (can be different from the actual signer).
+
+### Example
+
+<Tabs>
+<TabItem value="Request">
+
+```js
+await window.ethereum.request({
+  method: "wallet_invokeSnap",
+  params: {
+    snapId: "npm:@consensys/starknet-snap",
+    request: {
+      method: "starkNet_extractPublicKey",
+      params: {
+        userAddress: "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
+        chainId: "0x1"
+      },
+    },
+  },
+})
+```
+
+</TabItem>
+<TabItem value="Result">
+
+```json
+"0x04bfcab3b7ca7e8b3f3b62b2f7f77e9e4b68080bbf8f0f4a1c8f890864d2c7c1d3c45d8b2e3f5f1c27dfeea4c2f5733e90bfc7484e2a690aa9b8ac4559d2e6a8d7"
+```
+
+</TabItem>
+</Tabs>
+
+## `starkNet_getErc20TokenBalance`
+
+Gets the user's current balance of an ERC-20 token.
+
+### Parameters
+
+- `tokenAddress`: `string` - Address of the ERC-20 token contract.
+- `userAddress`: `string` - Address of the user account.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Goerli testnet.
+
+### Returns
+
+The token balance in hexadecimal.
```

Comment:
It returns an object with the pending and latest balance in hexadecimal : 

e.g.
  ```javascript
{
    balancePending: "0x0",
    balanceLatest: "0x0",
};
```